scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala

```diff
@@ -111,7 +120,29 @@ package object hdfs {
         schema
       }
       AvroJob.setOutputKeySchema(job, s)
-      val sink = new HadoopFileSink(path, classOf[AvroKeyOutputFormat[T]], job.getConfiguration)
+      val sink = new HadoopFileSink(path,
+                                    classOf[AvroKeyOutputFormat[T]],
+                                    job.getConfiguration)
+      self
+        .map(x => KV.of(new AvroKey(x), NullWritable.get()))
+        .applyInternal(Write.to(sink))
+      self.context.makeFuture(HdfsAvroTap[T](path, schema))
+    }
+
+    /** Save this SCollection as an Avro file on HDFS. */
+    // TODO: numShards
+    def saveAsHdfsAvroFile(path: String, schema: Schema, user: String): Future[Tap[T]] = {
```

Comment:
Same here, add `user: String = null` to the previous one. Also `schema` needs to be nullable too for `SpecificRecord`.